01_temperature_processing.R

```diff
@@ -185,31 +240,35 @@ for (i in 1:length(loggers)) {
                    tmp_data$DATETIME > obs.rt$DATETIME),"anomaly"] <- NA
   
   #Set up final grade column to be verified using shiny app and further review
-  tmp_data$DQL <- ifelse(tmp_data$field_audit_grade == 'C' | tmp_data$bath_grade == 'C',
-                         'C',
-                         ifelse(tmp_data$field_audit_grade == 'B' | tmp_data$bath_grade == 'B',
-                                'B',
-                                ifelse(tmp_data$field_audit_grade == 'A' & tmp_data$bath_grade == 'A',
-                                       'A',
-                                       ifelse(tmp_data$field_audit_grade == 'E' & tmp_data$bath_grade == 'E',
-                                              'E',
-                                              'B'))))
+  tmp_data$rDQL <- ifelse(tmp_data$field_audit_grade == 'C' | tmp_data$bath_grade == 'C',
+                          'C',
+                          ifelse(tmp_data$field_audit_grade == 'B' | tmp_data$bath_grade == 'B',
+                                 'B',
+                                 ifelse(tmp_data$field_audit_grade == 'A' & tmp_data$bath_grade == 'A',
+                                        'A',
+                                        ifelse(tmp_data$field_audit_grade == 'E' & tmp_data$bath_grade == 'E',
+                                               'E',
+                                               'B'))))
   
   #Set pre and post deployment as NA for future clipping
-  tmp_data$final_grade <- ifelse(tmp_data$dbf & tmp_data$raf, tmp_data$DQL, NA)
+  tmp_data$final_grade <- ifelse(tmp_data$dbf & tmp_data$raf, tmp_data$rDQL, NA)
+  
+  # anomaly check for provided dql not matching the calculated rDQL
+  tmp_data$anomaly <- ifelse(is.na(tmp_data$dql) | is.na(tmp_data$rDQL), tmp_data$anomaly, 
+                             ifelse(tmp_data$dql == tmp_data$rDQL, tmp_data$anomaly, FALSE))
   
   #Just keep the fields we want to persits
-  tmp_data <- tmp_data[,c('DATETIME', 'TEMP', 'DQL', 'anomaly', 
+  tmp_data <- tmp_data[,c('DATETIME', 'charid','r', 'dql','rDQL', 'anomaly', 
                           'field_audit_grade', 'bath_grade', 'daily_min', 
                           'daily_max', 'daily_mean', 'daily_diel')]
   
-  yr <- strftime(dr_info[1, 'AUDIT_DATETIME'], "%Y")
-  lasar <- smi[which(smi$"Logger ID" == loggers[i]), c('LASAR ID', "Station Description")]
-  fname <- paste(yr, loggers[i], lasar$"LASAR ID", lasar$"Station Description", ".Rdata", sep = "_")
-  fname <- gsub("/","_",fname)
+  dploy <- strftime(dr_info[1, 'AUDIT_DATETIME'], "%Y%m%d")
+  lasar <- smi[which(smi$Logger_ID == logchar[i,1]), c('LASAR_ID', "Station_Description")]
+  fname <- paste(SubID, lasar$LASAR_ID, logchar[i,1], logchar[i,2], dploy, lasar$Station_Description, ".Rdata", sep = "_")
+  fname <- gsub("/","_",fname) # not sure what this is for
```

Comment:
I think there were station descriptions that had a / character in them that was screwing up the filename